### ISSUE TITLE
Proposal: prefer-null-coalesce-assignment

### ISSUE THREAD

**Issue by boris-petrov** [eslint/eslint#14262]
**Please describe what the rule should do:**

Code like:

```javascript
x = x ?? 'default-value';
```

Could be written (and is "better"):

```javascript
x ??= 'default-value';
```

**What new ECMAScript feature does this rule relate to?**

Null coalesce assignment.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Well, the one in the beginning should be enough I guess.

**Why should this rule be included in ESLint (instead of a plugin)?**

Absolutely.

**Are you willing to submit a pull request to implement this rule?**

Not likely. :smile:

**Comment by mdjermanovic** [comment]
Hi @boris-petrov, thanks for the issue!

I think this is already included in proposal #13689, which also covers `&&=` and `||=`.

**Comment by boris-petrov** [comment]
@mdjermanovic - indeed it is, sorry I missed it! Thank you!
